### PR TITLE
Support Procs for url configuration

### PR DESCRIPTION
**Pull request by tjwp**
The intent is to use the Proc option to dynamically configure registry URLs for Heroku.

Prime: @jturkel

### CODE REVIEW

**Comment by tjwp** [comment]
@will89 Could you take over as prime on this? Joel looked at it before, but then I reworked things to allow environment variables to override the configuration on the module. I'm making this change because there's another gem (https://github.com/salsify/heroku_avrolution/pull/1) that will automatically set the configuration in development.

File: lib/avrolution/configuration.rb [hunk 1]
```diff
@@ -34,15 +47,11 @@ def self.compatibility_breaks_file
   end
 
   def self.compatibility_schema_registry_url
-    @compatibility_schema_registry_url ||= ENV.fetch(COMPATIBILITY_SCHEMA_REGISTRY_URL) do
-      raise 'compatibility_schema_registry_url must be set'
-    end
+    fetch_url('compatibility')
```
Comment (will89) [inline lib/avrolution/configuration.rb]: Should the fetch_url param be constant?

File: spec/avrolution/configuration_spec.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    context "when a value is not configured" do
+      let(:configured_value) { nil }
+
+      it "raises an error" do
+        expect { url }.to raise_error('compatibility_schema_registry_url must be set')
+      end
+    end
+
+    context "when the environment variable is set" do
```
Comment (will89) [inline spec/avrolution/configuration_spec.rb]: Might be worth modifying or add another test that makes sure the environment variable takes precedence over the assigned value.

File: lib/avrolution/configuration.rb (hunk 1 above)
Comment (tjwp) [inline lib/avrolution/configuration.rb]: changed

File: spec/avrolution/configuration_spec.rb (hunk 2 above)
Comment (tjwp) [inline spec/avrolution/configuration_spec.rb]: There is an assigned value on line 10 (https://github.com/salsify/avrolution/pull/6/files#diff-b357f6cd5e7a00c7f6afb2b01e6ff4dbR10) that this takes precedence over.

**Comment by will89** [comment]
Looks good to me 🚢

**Pull request merged**